src/meta/src/barrier/recovery.rs

```diff
@@ -64,8 +64,12 @@ where
         debug!("recovery start!");
         let retry_strategy = Self::get_retry_strategy();
         let (new_epoch, responses) = tokio_retry::Retry::spawn(retry_strategy, || async {
-            let info = self.resolve_actor_info(None).await;
+            let mut info = self.resolve_actor_info(None).await;
             let mut new_epoch = prev_epoch.next();
+
+            // Migrate expired actors to newly joined node by changing actor_map
+            self.migrate_actors(&mut info).await;
+
```

Comment:
After migration, we should persist the change to `MetaStore`, which requires `FragmentManager` to provide a transactional API.